### ISSUE TITLE
Asp dot net core bolt driver can't connect on *nix

### ISSUE THREAD

**Issue by ChaosYoda** [neo4j/neo4j-dotnet-driver#164]
When trying to run an app on *nix with dot net core, I get This platform does not support connecting sockets to DNS endpoints via the instance Connect and ConnectAsync methods, due to the potential for a host name to map to multiple IP addresses and sockets becoming invalid for use after a failed connect attempt. Use the static ConnectAsync method, or provide to the instance methods the specific IPAddress desired. This is related to https://github.com/dotnet/corefx/issues/8768 and https://github.com/StackExchange/StackExchange.Redis/issues/463. There is a work around but currently the issue seems to be in the driver.

**Comment by zhenlineo** [comment]
Thanks for reporting this issue. I was recently looking into resolving DNS to IPs before creating connection pool for this driver. I think your problem will be solved by introducing Dns resolving. I was holding doing it as I was a bit reluctant to introduce another async dns.ResolveAsync into this synchronized driver. 

While waiting for a fix, I would also suggest some work around for now. Consider to provide IP address directly to the driver rather than a Dns address. If you want to use multi IPs to avoid single point failure, then consider to put a load-balancer in the middle and pass the IP of load-balancer to the driver.

**Comment by ChaosYoda** [comment]
Hi, I am using the IP directly and it gives me the error above.

**Comment by zhenlineo** [comment]
Oh I see, so there is a problem with 
```
public Task ConnectAsync(string host, int port);
```

 and we should avoid the problem by moving to 

```
public Task ConnectAsync(IPAddress address, int port);
public Task ConnectAsync(IPAddress[] addresses, int port);
```

Am I got you correctly this time?

**Comment by ChaosYoda** [comment]
Yes I think that will work.

**Comment by zhenlineo** [comment]
Hi, the problem should go away with the new .net [1.3.0-beta01](https://github.com/neo4j/neo4j-dotnet-driver/releases/tag/1.3.0-beta01). The fix will also be included in 1.2.1 (which should be coming soon).
